buildpipeline/perf-pipeline.groovy

```diff
@@ -0,0 +1,390 @@
+    // submission-metadata.py
+    if (isBaseline) {
+        sh "mv submission-metadata-baseline.json submission-metadata.json"
+    }
+
+    sh "./tests/scripts/perf-prep.sh --throughput"
+    sh "./init-tools.sh"
+    sh "python3 ./tests/scripts/run-throughput-perf.py -arch \"${arch}\" -os \"${os}\" -configuration \"${config}\" -opt_level ${optLevel} -clr_root \"\${WORKSPACE}\" -assembly_root \"\${WORKSPACE}/${arch}ThroughputBenchmarks/lib\" -run_type \"${runType}\"  -benchview_path \"\${WORKSPACE}/tests/scripts/Microsoft.BenchView.JSONFormat/tools\""
+    archiveArtifacts allowEmptyArchive: false, artifacts:'throughput-*.csv,machinedata.json'
+}
+
+//-------------------------- Job Definitions --------------------------//
+
+String config = "Release"
+String runType = isPR() ? 'private' : 'rolling'
+
+// TODO: This needs to be updated before going live
+String uploadString = ''
+
+stage ('Get Metadata and download Throughput Benchmarks') {
+    simpleNode('Windows_NT', '20170427-elevated') {
+        checkout scm
+        String commit = getCommit()
+        def benchViewName = isPR() ? "coreclr private %ghprbPullTitle%" : "coreclr rolling %GIT_BRANCH_WITHOUT_ORIGIN% ${commit}"
+        def benchViewUser = getUserEmail()
+        bat "mkdir tools\n" +
+            "powershell Invoke-WebRequest https://dist.nuget.org/win-x86-commandline/v4.1.0/nuget.exe -OutFile %WORKSPACE%\\tools\\nuget.exe"
+        bat "%WORKSPACE%\\tools\\nuget.exe install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -Prerelease -ExcludeVersion"
+        bat "%WORKSPACE%\\tools\\nuget.exe install Microsoft.BenchView.ThroughputBenchmarks.x64.Windows_NT -Source https://dotnet.myget.org/F/dotnet-core -Prerelease -ExcludeVersion"
+        bat "%WORKSPACE%\\tools\\nuget.exe install Microsoft.BenchView.ThroughputBenchmarks.x86.Windows_NT -Source https://dotnet.myget.org/F/dotnet-core -Prerelease -ExcludeVersion"
+        bat "set \"GIT_BRANCH_WITHOUT_ORIGIN=%GitBranchOrCommit:*/=%\"\n" +
+            "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"${benchViewName}\" --user \"${benchViewUser}\"\n" +
+            "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch %GIT_BRANCH_WITHOUT_ORIGIN% --type ${runType}\n" +
+            "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"${benchViewName}-baseline\" --user \"${benchViewUser}\" -o submission-metadata-baseline.json\n"
+        
+        // TODO: revisit these moves. Originally, stash could not find the directories as currently named
+        bat "move Microsoft.BenchView.ThroughputBenchmarks.x64.Windows_NT x64ThroughputBenchmarks"
+        bat "move Microsoft.BenchView.ThroughputBenchmarks.x86.Windows_NT x86ThroughputBenchmarks"
+
+        stash name: 'benchview-artifacts', includes: 'Microsoft.BenchView.JSONFormat/**/*'
```

Comment:
nit: `benchview-artifacts` make me think of json files here. I think this is better off being called `benchview-tools` (and that is also how the JSONFormat scripts are called in the benchview sources proper).

Reply:
Sounds good